sycl/include/sycl/detail/cg.hpp

```diff
@@ -187,7 +187,8 @@ class CGExecKernel : public CG {
         MNDRDesc(std::move(NDRDesc)), MHostKernel(std::move(HKernel)),
         MSyclKernel(std::move(SyclKernel)),
         MKernelBundle(std::move(KernelBundle)), MArgs(std::move(Args)),
-        MKernelName(std::move(KernelName)), MStreams(std::move(Streams)),
+        MKernelName((KernelName != nullptr) ? strdup(KernelName) : nullptr),
```

Comment:
Why do we need to duplicate the string here? This seems to defeat the whole purpose of moving away from `std:string`.

Reply:
@sergey-semenov - The problem is string manages its own memory. If we dynamically allocate memory here for char* it will cause memory leak issues. How do you suggest we deal with it?